lib/worker.js

```diff
@@ -330,12 +344,39 @@ class AssetComputeWorker {
         }
     }
 
+    sendEventsBeforeActionTimeout() {
+        return setTimeout(
+            async () => {
+                console.log(`Action is about to timeout in ${timeUntilTimeout()}ms Sending IO events before timeout`);
+                // ensure failure events are sent for any non successful rendition before timeout
+                if(this.renditions){
+                    for (const rendition of this.renditions) {
+                        if (!rendition.eventSent) {
+                            await this.events.sendEvent(EVENT_RENDITION_FAILED, {
+                                rendition: rendition.instructionsForEvent(),
+                                errorReason: Reason.GenericError,
+                                errorMessage: "Timeout",
```

Comment:
Maybe add a bit more text: "Timed out without result after XXX seconds/minutes"

Leads to an interesting question: time from start of worker? That particular rendition might not have been started yet (another one or the download could have been the blocking issue).

Reply:
Currently it is the time from start of worker. Maybe to make it clearer we can say: "Worker timed out without result after XXX seconds/minutes"

Reply:
👍